vault/provider.go

```diff
@@ -705,6 +706,13 @@ func providerToken(d *schema.ResourceData) (string, error) {
 	return strings.TrimSpace(token), nil
 }
 
+func RecordMultipleStates(lastStates ...string) func(*api.Response) {
+	return func(resp *api.Response) {
+		newState := resp.Header.Get("X-Vault-Index")
+		lastStates = vault.MergeStates(lastStates, newState)
```

Comment:
May need locking here if the client is used by multiple goroutines.

Reply:
Included locking with a newly initialized mutex